docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,151 @@
+- 4. Develop or edit the command that you are working on
+- 5. Test your build locally (Instructions below)
+- 6. Create a PR to merge your command into staging (Template at bottom)
+
+# Testing your build
+
+### node and npm install
+
+- 1. [install node](https://nodejs.org/en/download/), npm is bundled with the download.
+- 2. Open a command prompt or terminal in your repo directory and type `npm install`.
+
+### Bot application
+
+- 1. Log into the Discord website and navigate to the [applications page](https://discord.com/developers/applications)
+- 2. Click `New Application`
+- 3. Name your application
+- 4. Navigate to the `Bot` tab and click `Add Bot`. You will have to confirm by clicking `Yes, do it!`
+- 5. Click the `Copy` button underneath token. (Do not share your token with anyone)
+- 6. Create a file called `.env` in your repo
+- 7. Inside the `.env` file, type `DISCORD_TOKEN=TOKEN` replacing token with what you copied in step 5
+- 8. You may need to add the .env file to the `gitignore` if your IDE has not done so automatically
+
+### Privilaged Gateway Intents
+
+Privileged Gateway Intents must now be enabled within the Discord Developer Portal in order for your bot to function.
+
+follow the steps below to enable them
+
+- 1. log into the Discord webiste and navigate to the [applications page](https://discord.com/developers/applications) then select `Bot` under `settings`
+- 2. Scroll down to the Privilaged Gateway Intents section and enable all the intents.
+
+### Inviting the Bot to your server
+
+- 1. Create a `private` server where you are able to freely test your Bot
+- 2. on the [applications page](https://discord.com/developers/applications) select your application and navigate to the `Oauth2` tab. Then select `bot` under the `scopes` section
+- 3. Tick `Administrater` under the `Bot Permissions` section
+- 4. click the `copy` button and paste it into your browser of choice, invite it to your test server.
+
+### Running the Bot
+
+- 1. open a command prompt/terminal in your repo directory
+- 2. Run `npm run dev`
+- 3. If all has gone well, you will see the Bot is running as 'http://localhost:3000' and logged into the name of the Bot you created!
+- 4. You can now test your commands
+
+# Editing Methods
+
+### Adding a new command
+
+- 1. Create a new file in the relevant folder `src/commands/` and name it appropriately. `yourcommand.ts`
+- 2. Add the licence agreement
+
+```ts
+//============================================================
+//
+// Salty Simulations Discord Bot
+// Author: saltysimulations
+// URL: https://github.com/owen2007/Salty-Simulations-Discord-Bot/
+// Date: 2022-04-25
+// Version: 0.0.1
+// License: AGPLv3
+//
+// Source based on:
+// FlyByWire Discord Bot
+// Author: flybywire
+// URL: https://github.com/flybywiresim/discord-bot/
+//
+//============================================================
+//
+//
+//
+//============================================================
+```
```

Comment:
Mention somewhere around here that the date should be updated to the date of file creation